news/7094.removal

```diff
@@ -0,0 +1 @@
+Remove DependencyWarning warning from pip._internal
```

Comment:
I would make this a .trivial file since it is a refactoring that should not have any user impact.

Reply:
Oh alright. Let me change that